task1.c

```diff
@@ -0,0 +1,118 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <stdbool.h>
+
+void swap(int *firstElement, int *secondElement)
+{
+    int transitElement = *firstElement;
+    *firstElement = *secondElement;
+    *secondElement = transitElement;
+}
+
+void printArray(int *array, int size)
+{
+    for (int i = 0; i < size; ++i)
+    {
+        printf("%d ", array[i]);
+    }
+    printf("\n");
+}
+
+int elementsPermutation(int *array, int arrayLength)
+{
+    int counterOfPermutations = 0;
+    int j = 0;
+    for (int i = 0; i < arrayLength; ++i)
+    {
+        if (array[i] % 2 == 0)
+        {
+            if (array[i - 1] % 2 == 0 || i == 0)
+            {
+                ++j;
+            }
+            else
+            {
+                swap(&array[i], &array[j++]);
+                ++counterOfPermutations;
+            }
+        }
+    }
+    return counterOfPermutations;
+}
+
+bool test1(void)
+{
+    int arrayBeforePermutations[5] = {2, 1, 3, 4, 5};
+    int arrayAfterPermutations[5] = {2, 4, 3, 1, 5};
+    elementsPermutation(arrayBeforePermutations, 5);
+    for (int i = 0; i < 5; ++i)
+    {
+        if (arrayAfterPermutations[i] != arrayBeforePermutations[i])
+        {
+            return false;
+        }
+    }
+    return true;
+}
+
+bool test2(void)
+{
+    int arrayBeforePermutations[5] = {2, 4, 3, 1, 5};
+    int arrayAfterPermutations[5] = {2, 4, 3, 1, 5};
+    int countOfPermutations = elementsPermutation(arrayBeforePermutations, 5);
+    if (countOfPermutations == 0)
+    {
+        return true;
+    }
+    return false;
```

Comment:
```suggestion
    return countOfPermutations == 0;
```
Если true, то верни true, иначе верни false. Тавтология же.